core/transcoder.go

```diff
@@ -77,6 +152,79 @@ func (nv *NvidiaTranscoder) Transcode(job string, fname string, profiles []ffmpe
 	return resToTranscodeData(res, out)
 }
 
+func testOneNvidiaDevice(fname, gpu string, start chan interface{}, res chan error) {
+	<-start
+	t1 := NewNvidiaTranscoder(gpu)
+	// t1 := NewLocalTranscoder("/tmp")
+	profiles := []ffmpeg.VideoProfile{ffmpeg.P720p60fps16x9}
+	profiles[0].Format = ffmpeg.FormatMP4
+	// fname = "rendtest.mp4"
+	td, err := t1.Transcode("", fname, profiles)
+
+	t1.Stop()
+	if err != nil {
+		res <- err
+		return
+	}
+	if len(td.Segments) == 0 || td.Pixels == 0 {
+		res <- errors.New("Empty transcoded segment")
+		return
+	}
+	res <- nil
+}
+
+// TestNvidiaTranscoder tries to transcoded test segment on all the devices
+func TestNvidiaTranscoder(gpu string) error {
+	devices := strings.Split(gpu, ",")
+	b := bytes.NewReader(testSegment)
+	z, err := gzip.NewReader(b)
+	if err != nil {
+		return err
+	}
+	mp4testSeg, err := ioutil.ReadAll(z)
+	z.Close()
+	if err != nil {
+		return err
+	}
+	tmp, err := ioutil.TempDir("", "")
+	if err != nil {
+		return err
+	}
+	if true {
+		defer os.RemoveAll(tmp)
+	}
+	fname := path.Join(tmp, "testseg.mp4")
```

Comment:
We have a `WorkDir`, might be more consistent to use that. Also, I believe `filepath` is recommended when dealing with files, so:
 
```
fname = filepath.Join(WorkDir, "test.tempfile")
defer os.Remove(fname)
```